### DISCUSSION TITLE
Asking to use "Zigbee Model" and "Zigbee Manufacturer" when submitting Zigbee2mqtt integrated devices?

### DISCUSSION THREAD

**Discussion by jpbaril** [andrew-codechimp/HA-Battery-Notes#2014] [Ideas]
I began using this integration recently and even more recently I added a second zigbee coordinator that use Zigbee2mqtt. So I now have some zigbee devices integrated through ZHA and some through Z2M.

In Home Assistant devices that are integrated though MQTT are listed with some friendly/human-comprehensible model and manufacturer names that come from the Zigbee2MQTT devices database. They are not the "raw" models and manufacturers.
On the other hand, if a device is integrated through ZHA the listed models and manufactures will be those "raw" values.

But, those "raw" models/manufactures are also shown in Z2M itself.
They are called "Zigbee Model" and "Zigbee Manufacturer".

Ex:
- ZHA Model / Z2M Zigbee Model : TS0222
- ZHA Manufacturer / Z2M Zigbee Manufacturer :  _TZ3000_kky16aay
- Z2M Model : QT-07S
- Z2M Manufacturer : Tuya

So, as someone else already [pointed out](https://github.com/andrew-codechimp/HA-Battery-Notes/issues/88), your Battery-Notes database is "doomed" to contain duplicates for every single device.

Thus, to stop the duplicates, since Zigbee2MQTT is more for "advanced" users, I think it would make sense, would not confuse people, and would not be a burden if you requested people to use "Zigbee Model" and "Zigbee Manufacturer" when submitting new devices for inclusion in your database.

BTW, in that example the Z2M "QT-07S" model is wrong, it should be "TS0222_temperature_humidity". This will be corrected in next Z2M release. Another good reason to instead use the "raw" data.

**Comment by andrew-codechimp** [comment]
To be able to detect a device I have to use the [Device Entry](https://developers.home-assistant.io/docs/device_registry_index/) data available from HA Core that is standardised for all devices.  It is up to the integration developer what is presented here so I'm at the whim of implementations which is frustrating at times, serial numbers in model is one recent one which I had to introduce wildcards for.

As well as ZHA and Z2M there's also the other zigbee hubs to consider like Hue and Homekit which also present different information.  If I were to start looking at interrogating by a specific integration that would be a lot more effort than maintaining duplicates, and whilst I have a lot of devices I don't have all the different integrations available to test with, and that's just Zigbee, there's Z-Wave, WiFi local, WiFi Cloud for things like Tuya with probably just as much duplication.

It's also hard getting the right data submitted, particularly with the new Model ID in core, the next release currently in beta will improve that a little, having users submit data from different sources depending on integration will just make it harder. 

There is some thought going on within the core team on device databases but that's very early days, the recently added Model ID is a step towards getting things more organised so I'll be watching that closely to see if it can improve my library.

We seem to have reached near peak device library volumes now, I get a few a week and will probably continue to do so as new devices come out but not the 10+ a day that I had when I first started this, the library lookups are still performant and a daily pull of the library isn't huge bandwidth wise.

The example you gave of the Z2M "QT-07S" model being wrong, that will require a duplication as not everyone updates to the latest Z2M as quickly as probably you or me.
